Sources/Profile-Modify.php

```diff
@@ -1193,13 +1193,29 @@ function makeCustomFieldChanges($memID, $area, $sanitize = true, $returnErrors =
 		else
 		{
 			$value = isset($_POST['customfield'][$row['col_name']]) ? $_POST['customfield'][$row['col_name']] : '';
+
 			if ($row['field_length'])
 				$value = $smcFunc['substr']($value, 0, $row['field_length']);
 
 			// Any masks?
 			if ($row['field_type'] == 'text' && !empty($row['mask']) && $row['mask'] != 'none')
 			{
-				if ($row['mask'] == 'email' && (!filter_var($value, FILTER_VALIDATE_EMAIL) || strlen($value) > 255))
+				$value = $smcFunc['htmltrim']($value);
+				$valueReference = un_htmlspecialchars($value);
+
+				// Try and avoid some checks. '0' could be a valid non-empty value.
+				if (empty($value) && !is_numeric($value))
+					$value = '';
+
+				if ($row['mask'] == 'nohtml' && ($valueReference != strip_tags($valueReference) || $value != filter_var($value, FILTER_SANITIZE_STRING) || preg_match('/<(.+?)[\s]*\/?[\s]*>/si', $valueReference)))
+				{
+					if ($returnErrors)
+						$errors[] = 'custom_field_nohtml_fail';
+
+					else
+						$value = '';
+				}
+				elseif ($row['mask'] == 'email' && (!filter_var($value, FILTER_VALIDATE_EMAIL) || strlen($value) > 255))
```

Comment:
`!filter_var($value, FILTER_VALIDATE_EMAIL)` -> `$value != filter_var($value, FILTER_VALIDATE_EMAIL)`

Reply:
Well, I didn't modified that part but its actually correct.  Theres a difference between VALIDATE and SANITIZE options.
For the nohtml mask there is no VALIDATE_HTML_SOMETHING filter which is why we indeed have to use a comparison check !=

For the mail mask however, FILTER_VALIDATE_EMAIL will return false if the filter fails which is what that piece of code is checking against. 

So, doing !filter_var($value, FILTER_VALIDATE_EMAIL) is the same as $value != filter_var($value, FILTER_VALIDATE_EMAIL)

Reply:
Thank you... missed this.